tests/phpunit/includes/testcase-domain.php

```diff
@@ -1,27 +1,113 @@
 <?php
 
 class PLL_Domain_UnitTestCase extends PLL_UnitTestCase {
-	protected $structure = '/%postname%/';
+	use PLL_Links_Trait;
+
 	protected $hosts;
-	protected static $server;
+	protected $is_directory = false;
 
 	/**
 	 * @param WP_UnitTest_Factory $factory
 	 */
 	public static function wpSetUpBeforeClass( WP_UnitTest_Factory $factory ) {
 		parent::wpSetUpBeforeClass( $factory );
 
-		self::$server = $_SERVER; // backup
-
 		self::create_language( 'en_US' );
 		self::create_language( 'fr_FR' );
 		self::create_language( 'de_DE' );
+
+		self::$model->options['default_lang'] = 'en';
+	}
+
+	public function set_up() {
+		parent::set_up();
+
+		$this->filter_plugins_url();
+	}
+
+	public function tear_down() {
+		parent::tear_down();
+
+		$this->reset__SERVER();
+	}
+
+	protected function base_test_flags_urls( $curlang ) {
+		// Needed by {@see pll_requested_url()}.
+		$_SERVER['HTTP_HOST'] = wp_parse_url( $this->hosts[ $curlang->slug ], PHP_URL_HOST );
+
+		$frontend          = new PLL_Frontend( $this->links_model );
+		$frontend->curlang = $curlang;
+		$frontend->init();
+		$languages = $frontend->model->get_languages_list();
+
+		$this->assertCount( 3, $languages ); // @see `self::wpSetUpBeforeClass()`.
+
+		foreach ( $languages as $flag_language ) {
+			$code = 'en' === $flag_language->slug ? 'us' : $flag_language->slug;
+			$dir  = $this->is_directory ? '/sub' : '';
+			$this->assertSame(
+				$this->hosts[ $curlang->slug ] . "{$dir}/wp-content/plugins/polylang/flags/{$code}.png",
+				$flag_language->get_display_flag_url(),
+				"{$flag_language->name} flag URL with current language set to {$curlang->name} is wrong."
+			);
+		}
+	}
+
+	/**
+	 * Enables a WordPress installation in directory.
+	 *
+	 * @return void
+	 */
+	protected function maybe_set_directory() {
+		if ( ! $this->is_directory ) {
+			return;
+		}
+
+		// Fake WP install in subdir.
+		update_option( 'siteurl', 'http://example.org/sub' );
+		update_option( 'home', 'http://example.org' );
 	}
 
-	public static function wpTearDownAfterClass() {
-		parent::wpTearDownAfterClass();
+	/**
+	 * @ticket #1296
+	 * @see https://github.com/polylang/polylang/issues/1296.
+	 */
+	public function test_flags_urls_curlang_default() {
```

Comment:
I suggest to use a provider to test WordPress installed at root or in a subfolder. This should avoid duplicate code.

Reply:
Done in a5d7ac57.